src/main/java/com/jeovan/gymcrmsystem/daos/TraineeDao.java

```diff
@@ -1,15 +1,22 @@
 package com.jeovan.gymcrmsystem.daos;
 
 import com.jeovan.gymcrmsystem.models.Trainee;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
 
-import java.util.Map;
+import java.util.Optional;
 import java.util.UUID;
 
 @Component
-public class TraineeDao extends SimpleDao<Trainee>{
+public interface TraineeDao extends JpaRepository<Trainee, UUID> {
 
-    public TraineeDao(Map<UUID, Trainee> storage) {
-        super(storage);
-    }
+    @Query("SELECT t FROM Trainee t JOIN t.user u WHERE u.username = ?1")
+    Optional<Trainee> findByUsername(String username);
+    @Modifying
+    @Transactional
+    @Query("DELETE FROM Trainee t WHERE EXISTS (SELECT 1 FROM User u WHERE u.username = ?1 AND u = t.user)")
+    int deleteByUsername(String username);
```

Comment:
same comment here, as we are using spring jpa perhaps we can just use query name method to execute this delete operation and get rid @modifyng and @query

Reply:
Same as the other one, fixed!